responsibleai/responsibleai/_internal/_forecasting_wrappers.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+# All wrappers in this file will be migrated to ml-wrappers.
+
+import inspect
+import sys
+
+import pandas as pd
+from ml_wrappers.model.base_wrapped_model import BaseWrappedModel
+
+from responsibleai.rai_insights.constants import ModelTask
+
+from .constants import _Forecasting
+
+_AZUREML = "azureml"
+_SKTIME = "sktime"
+
+
+def _get_model_package(model):
+    module = inspect.getmodule(model)
+    module_name = module.__name__.partition('.')[0]
+    model_package = sys.modules[module_name].__package__
+    return model_package
+
+
+def _wrap_model(model, examples, time_feature, time_series_id_features):
+    """Choose the appropriate wrapper for the model.
+
+    If the model has a forecast_quantiles method
+    (or in the sktime case: predict_quantiles) then choose the wrapped
+    quantile forecasting model, otherwise the basic wrapper without
+    quantile forecasting capabilities.
+    """
+    model_package = _get_model_package(model)
+    if ((model_package == _SKTIME and hasattr(model, "predict_quantiles")) or
+            hasattr(model, _Forecasting.FORECAST_QUANTILES)):
+        return _WrappedQuantileForecastingModel(
+            model, examples, time_feature, time_series_id_features)
+    else:
+        return _WrappedForecastingModel(
+            model, examples, time_feature, time_series_id_features)
+
+
+class _WrappedForecastingModel(BaseWrappedModel):
+    def __init__(self, model, examples, time_feature,
+                 time_series_id_features):
+        super(_WrappedForecastingModel, self).__init__(
+            model,
+            eval_function=None,
+            examples=examples,
+            model_task=ModelTask.FORECASTING)
+        self._model_package = _get_model_package(model)
+        if (self._model_package not in [_AZUREML, _SKTIME] and
+                not hasattr(model, _Forecasting.FORECAST)):
+            raise ValueError(
+                "The passed model does not have a 'forecast' method. "
+                "'forecast' is required for the forecasting task_type. "
+                "Alternatively, pass a sktime forecasting model.")
+        self._validate_time_features(time_feature, time_series_id_features)
+
+    def _validate_time_features(self, time_feature, time_series_id_features):
+        if time_feature not in self._examples.columns:
+            raise ValueError(
+                "The passed time_feature is not in the examples DataFrame.")
+        self._time_feature = time_feature
+        for ts_id in time_series_id_features:
+            if ts_id not in self._examples.columns:
+                raise ValueError(
+                    f"The passed time series ID feature {ts_id} is not "
+                    "in the examples DataFrame.")
+        self._time_feature = time_feature
+        self._time_series_id_features = time_series_id_features
+
+    def forecast(self, X):

```

Comment:
is this the API we want to go with instead of predict?

Reply:
perhaps we can also have both, with predict an alias for forecast?

Reply:
I had a set of discussions about this with the forecasting team. They strongly recommend using forecast rather than predict. That said, you can easily wrap your model with predict function inside the wrapper. Does that make sense to you?

I'm a bit reluctant to have one as an alias for the other. A model might have both functions, and we can hardly ask users to remove one of them if they're not identical. Or they might use yet another method. Or worse, they have these and one works differently from the other. In summary, it opens up a few additional failure cases without really adding anything. I'd rather have people consciously choosing to provide forecast (with a wrapper if need be). I'll provide an example notebook so it's clear.